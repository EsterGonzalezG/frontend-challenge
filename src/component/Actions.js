import React from 'react';

export const Actions = ({ product, setColor, setStorage }) => (
  <div>
    {
      <form action=''>
        <div className='color_container'>
          <label htmlFor='color'>Color:</label>
          <select name='color' id='color' value={product.colors} onChange={(event) => setColor(event.target.value)}>
            {product.options
              ? product.options.colors.map((color, index) =>
                  index === 0 ? (
                    <option value={color.code} key={index} defaultValue={color.name}>
                      {color.name}
                    </option>
                  ) : (
                    <option value={color.code} key={index}>
                      {color.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
        <div className='storage_container'>
          <label htmlFor='storage'>Almacenamiento:</label>
          <select
            name='storage'
            id='storage'
            value={product.storage}
            onChange={(event) => setStorage(event.target.value)}
          >
            {product.options
              ? product.options.storages.map((storage, index) =>
                  index === 0 ? (
                    <option value={storage.code} key={index} defaultValue={storage.name}>
                      {storage.name}
                    </option>
                  ) : (
                    <option value={storage.code} key={index}>
                      {storage.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
      </form>
    }
    <button>AÃ±adir</button>
  </div>
);
