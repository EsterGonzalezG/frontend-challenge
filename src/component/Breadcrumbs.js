import React from 'react';
import { Link, useLocation } from 'react-router-dom';

export const Breadcrumbs = React.memo(() => {
  const location = useLocation();
  const config = {
    home: [{ label: 'Home', path: '/' }],
    details: [{ label: 'Home', path: '/' }, { label: 'Product details' }],
  };

  function crumb() {
    if (location.pathname === '/') {
      return renderCrumb(config.home);
    } else {
      return renderCrumb(config.details);
    }
  }

  function renderCrumb(option) {
    return option.map(({ label, path }, index) => {
      return option.length - 1 === index ? (
        <li className='breadcrumbs-item' key={index}>
          {label}
        </li>
      ) : (
        <li key={index}>
          <Link to={path} className='breadcrumbs-link'>
            {label}
          </Link>
        </li>
      );
    });
  }

  return (
    <nav>
      <ul className='breadcrumbs color-dark-60'>{crumb()}</ul>
    </nav>
  );
});
