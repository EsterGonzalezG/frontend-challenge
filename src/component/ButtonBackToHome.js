import { PropTypes } from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

export const ButtonBackToHome = ({ children }) => (
  <Link className='' to='/'>
    {children}
  </Link>
);
ButtonBackToHome.prototype = {
  children: PropTypes.string.isRequired,
};
