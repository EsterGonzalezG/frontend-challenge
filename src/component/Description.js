import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { setTitleDescription, Subtitle } from './index';

export const Description = React.memo(({ description }) => {
  const [descriptionValue, setDescriptionValue] = useState([]);
  useEffect(() => {
    setDescriptionValue(setTitleDescription(description));
  }, [description]);

  return (
    <>
      <Subtitle>Characteristics:</Subtitle>
      <ul className='color-dark-100 l-marginBottom-24 font-m'>
        {descriptionValue.map(({ id, value }) => (
          <li key={id} className='color-violet'>
            {id} :
            {value !== '' ? <span className='color-dark-60'>{value}</span> : <span className='color-dark-60'> - </span>}
          </li>
        ))}
      </ul>
    </>
  );
});

Description.prototype = {
  description: PropTypes.shape({
    brand: PropTypes.string,
    model: PropTypes.string,
    price: PropTypes.string,
    cpu: PropTypes.string,
    ram: PropTypes.string,
    os: PropTypes.string,
    displayResolution: PropTypes.string,
    battery: PropTypes.string,
    primaryCamera: PropTypes.string,
    secondaryCmera: PropTypes.string,
    dimentions: PropTypes.string,
    weight: PropTypes.string,
  }).isRequired,
};
