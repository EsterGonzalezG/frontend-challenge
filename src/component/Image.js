import React from "react";
import PropTypes from 'prop-types';


export const Image = ({ item }) => (
    <img loading="lazy" className='l-marginBottom-16' src={item.imgUrl} alt={item.title}/>
)

Image.prototype = {
    imgUrl: PropTypes.string,
    title: PropTypes.string
}