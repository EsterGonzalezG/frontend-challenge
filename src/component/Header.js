import React from 'react';

import { Title } from './Title';
import { Breadcrumbs } from './Breadcrumbs';
import {ItemCard}  from './ItemCart'
export const Header = () => (
    <div className='header l-marginBottom-24'>
        <div className='header-content'>
        <Title>Tienda de mÃ³biles</Title>
        <Breadcrumbs/>
        </div>
        <ItemCard/>
    </div>
)


