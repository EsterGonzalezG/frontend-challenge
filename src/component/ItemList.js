import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { Image } from './Image';
import { NotData } from './NotData';
import { Search } from './Search';

export const ItemList = ({ list }) => {
  const [valueSearch, setValueSearch] = useState('');

  const [phones, setPhones] = useState([]);

  useEffect(() => {
    setPhones(list);
  }, [list]);

  if (list) {
    return (
      <div className='itemList l-content-wide'>
        <Search setValueSearch={setValueSearch} />
        <ul className='itemList-list'>
          {phones
            ?.filter(
              (product) =>
                product.brand.toLowerCase().includes(valueSearch.toLocaleLowerCase()) ||
                product.model.toLowerCase().includes(valueSearch.toLocaleLowerCase()),
            )
            .map((product, index) => (
              <li key={index} className='Card l-marginBottom-24'>
                <Link to={`/detail/${product.id}`}>
                  <div className='itemList-element'>
                    <div className='itemList-img l-marginBottom-16'>
                      <Image image={product.imgUrl} />
                    </div>
                    <ul>
                      <li key={product.brand}>
                        Brand : <span className='color-dark-60'>{product.brand}</span>
                      </li>
                      <li key={product.model}>
                        Model :<span className='color-dark-60'> {product.model}</span>
                      </li>
                      <li key={product.price}>
                        Price:{' '}
                        {product.price !== '' ? (
                          <span className='color-dark-60'>{product.price} </span>
                        ) : (
                          <span className='color-dark-60'> - </span>
                        )}
                      </li>
                    </ul>
                  </div>
                </Link>
              </li>
            ))}
        </ul>
      </div>
    );
  } else {
    <NotData>Sorry, at the moment there is no data to display </NotData>;
  }
};

ItemList.prototype = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      brand: PropTypes.string,
      model: PropTypes.string,
      price: PropTypes.string,
    }).isRequired,
  ),
};
