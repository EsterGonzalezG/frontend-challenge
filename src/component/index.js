import { setTitleDescription } from '../utils/setTitleDescription';
import { CartContext } from './../useContext/CartContext';
import { Breadcrumbs } from './Breadcrumbs';
import { ButtonBackToHome } from './ButtonBackToHome';
import { Image } from './Image';
import { ItemCard } from './ItemCart';
import { NotData } from './NotData';
import { Search } from './Search';
import { Select } from './Select';
import { Title } from './Title';

export {
  ButtonBackToHome,
  Select,
  Breadcrumbs,
  ItemCard,
  Image,
  Search,
  CartContext,
  NotData,
  Title,
  setTitleDescription,
};
