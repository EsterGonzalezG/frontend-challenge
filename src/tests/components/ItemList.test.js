import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { ItemList } from '../../component/ItemList';

describe('Test Item List component', () => {
  const list = [];
  const loading = true;
  const wrapper = shallow(<ItemList list={list} loading={loading} />);
  test('should be render a component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should be show loading', () => {
    const spinner = wrapper.find('div').text().trim();
    expect(spinner).toBe('LOADING...');
  });
  test('should be component not data', () => {
    const loading = false;
    const wrapper = shallow(<ItemList list={list} loading={loading} />);
    const componentNotData = wrapper.find('NotData');
    expect(componentNotData.exists()).toBe(true);
  });
  test('should be not show component not data and show link component', () => {
    const list = [{ id: 123, imgUrl: 'https://localhost/imagen.jpg', brand: 'alcatel', model: 'model', price: 120 }];
    const loading = false;
    const wrapper = shallow(<ItemList list={list} loading={loading} />);
    const componentNotData = wrapper.find('NotData');
    const link = wrapper.find('Link');
    expect(componentNotData.exists()).toBe(false);
    expect(link.exists()).toBe(true);
  });
});
