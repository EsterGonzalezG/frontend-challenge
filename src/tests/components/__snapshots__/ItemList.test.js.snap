// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Item List component should be render a component 1`] = `
<div>
  LOADING...
</div>
`;
